### PR TITLE
Implement `construct_runtime!` macro

### PR DESCRIPTION
**Pull request by bkchr**
The macro currently does not support renaming of types. So only `Module`, `Call`, etc are required.

Progress on: https://github.com/paritytech/substrate/issues/669

### CODE REVIEW

**Comment by bkchr** [comment]
@gavofyork now it also compiles :)

File: node/runtime/src/lib.rs [hunk 1]
```diff
@@ -17,6 +17,7 @@
 //! The Substrate runtime. This can be compiled with ``#[no_std]`, ready for Wasm.
 
 #![cfg_attr(not(feature = "std"), no_std)]
+#![recursion_limit="256"]
```
Comment (pepyakin) [inline node/runtime/src/lib.rs]: Can you please answer (prefferable leaving a comment), why it's increased (like what dependency requires it)?

File: srml/support/src/runtime.rs [hunk 2]
```diff
@@ -0,0 +1,999 @@
...
+		);
+	};
+	(
+		$runtime:ident;
+		$( $parsed_modules:ident :: $parsed_name:ident ),*;
+		;
+	) => {
+		mashup! {
+			$(
+                substrate_generate_config_name["config-name" $parsed_name] = $parsed_name Config;
```
Comment (pepyakin) [inline srml/support/src/runtime.rs]: Few spaces

File: srml/support/src/runtime.rs [hunk 3]
```diff
@@ -0,0 +1,999 @@
...
+			)*
+        }
```
Comment (pepyakin) [inline srml/support/src/runtime.rs]: Few spaces

**Pull request merged**